src/pmdas/openmetrics/pmdaopenmetrics.python

```diff
@@ -601,7 +603,13 @@ class Source(object):
         '''
         now = time.time()
         last_try_age = now - self.refresh_time
-        return len(self.metrics_by_name) == 0 or last_try_age > self.pmda.refresh_timeout
+
+        if len(self.metrics_by_name) == 0 and self.first_refresh is True:
+            refresh = True
+        else:
+            refresh = last_try_age > self.pmda.refresh_timeout
+
+        return refresh
```

Comment:
This change would be a little simpler not adding a new 'refresh' variable...?  (also don't think "is True" is needed)

```
        if len(self.metrics_by_name) == 0 and self.first_refresh:
            return True
        return last_try_age > self.pmda.refresh_timeout
```

Also, hmm, isn't  len(self.metrics_by_name) always zero on the first refresh?